src/finch/symbolic/term.py

```diff
@@ -49,8 +49,10 @@ def __hash__(self) -> int:
             )
         return self._hashcache
 
-    def __eq__(self, other: "Term") -> bool:
-        self.head() == other.head() and self.children() == other.children()
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, Term):
```

Comment:
`other` should be an `object`; and if it's incompatible, `NotImplemented` should be returned here.

Reply:
Sounds good.